src/common/io_util.cc

```diff
@@ -151,9 +151,7 @@ Status SockConnect(const std::string &host, uint32_t port, int *fd, int conn_tim
     sin.sin_port = htons(port);
 
     fcntl(*fd, F_SETFL, O_NONBLOCK);
-    if (connect(*fd, reinterpret_cast<sockaddr *>(&sin), sizeof(sin))) {
-      return Status::FromErrno();
-    }
+    connect(*fd, reinterpret_cast<sockaddr *>(&sin), sizeof(sin));
```

Comment:
I think maybe we can filter some specific error codes, e.g.
```cpp
if (int err = connect(...); err && errno != EINPROGRESS)
```

Reply:
Is it enough for `Util::aeWait()` on the next line to handle the error code?

Reply:
Looks good to me.